storage/innobase/fil/fil0crypt.cc

```diff
@@ -1428,8 +1487,9 @@ static bool fil_crypt_space_needs_rotation(rotate_thread_t *state,
       break;
     }
 
-    /* No need to rotate space if encryption is disabled */
-    if (crypt_data->is_encryption_disabled()) {
+    /* No need to rotate space if encryption is disabled
+     * permanently or temporarily */
+    if (crypt_data->is_encryption_disabled() || space->exclude_from_rotation) {
```

Comment:
Like space->crypt_data, this variable (space->exclude_from_rotation) is also inconsistent w.r.t mutexes.

Sometimes, it is read/write under fil_crypt_threads_mutex, sometimes no mutex.. :(

Reply:
This variable is std::atomic<bool> so no mutex would be required.